File: volume1/chapter8/ruby_s3_example/s3_example.rb
require 'rubygems'
require 'bundler/setup'
require 'aws-sdk-s3'
require 'pathname'

s3 = Aws::S3::Client.new(region: 'us-east-1')

# Does bucket exist?
bucket_name = "test1.replacewithyourdomain.com"
list_results = s3.list_buckets()
bucket_exists = false
for b in list_results.buckets do
    if b.name == bucket_name
        bucket_exists = true
        break
    end
end

# Create bucket if it does not exist
bucket_exists = false
if !bucket_exists
  create_results = s3.create_bucket({
      acl: "public-read",
      bucket: bucket_name,
    })
end

# Upload file
file_path = "path_to_your_file"
file_name = File.basename(file_path)

put_results = s3.put_object({
  body: file_path, 
  bucket: bucket_name, 
  key: file_name, 
})

File: volume1/chapter8/python_s3_example/s3_example.py
import boto3
import os

# Create a S3 client
s3 = boto3.client('s3')
response = s3.list_buckets()
buckets = response['Buckets']
bucket_exists = False
bucket_name = "test1.replacewithyourdomain.com"
for b in buckets:
    bname = b['Name']
    if "test1.replacewithyourdomain.com" in bname:
        bucket_exists = True
        break
        
# Create bucket if it does not exist.
if not bucket_exists:
    response = s3.create_bucket(
        Bucket=bucket_name,
        ACL='public-read'
    )
    print("Created  ", response['Location'])


# Upload to bucket.   
# Replace pathname with path to your file.  
pathname = "path_to_your_file"
key = os.path.basename(pathname)
s3.upload_file(pathname, bucket_name, key)
